legate/tester/stages/_linux/gpu.py

```diff
@@ -67,6 +67,15 @@ def shard_args(self, shard: Shard, config: Config) -> ArgList:
                 "--ranks-per-node",
                 str(config.ranks),
             ]
+        if config.nodes > 1:
+            args += [
+                "--nodes",
+                str(config.nodes),
+            ]
+        if config.launcher != "none":
+            args += ["--launcher", str(config.launcher)]
+        for extra in config.launcher_extra:
+            args += ["--launcher-extra=" + str(extra)]
```

Comment:
This same block is repeated a few times, maybe factor it to a helper on the base, then
```
args += self._handle_multi_node_args(config)
```
or similar?